sorting/selection_sort_recursive.c

```diff
@@ -0,0 +1,102 @@
+/**
+ * @file
+ * @author [Dhruv Pasricha](https://github.com/DhruvPasricha)
+ * @brief [Selection Sort](https://en.wikipedia.org/wiki/Selection_sort)
+ * implementation using recursion.
+ */
+#include <assert.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <time.h>
+
+/**
+ * @brief Swapped two numbers using pointer
+ * @param first pointer of first number
+ * @param second pointer of second number
+ */
+void swap(int *first, int *second)
```

Comment:
Consider using `uint64_t`, `uint32_t`, `uint16_t`, or even `uint8_t` for non-negative values. For negative values, use `int64_t` or their appropriate size ([reference](https://en.cppreference.com/w/c/types/integer)). Requires the `inttypes.h` header. Check out other parts of the code, too. 🙂

Reply:
Okay

Reply:
As all the values are non negative and can easily fit in 8 bits.
So, I have replaced int by uint8_t.